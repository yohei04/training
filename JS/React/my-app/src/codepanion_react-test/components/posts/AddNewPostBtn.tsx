import React from 'react';

interface AddNewPostBtnProps {
  onClick: () => void;
}

const AddNewPostBtn: React.FC<AddNewPostBtnProps> = ({ onClick }) => {
  return <button onClick={onClick}>Add New Post</button>;
};

export default AddNewPostBtn;
