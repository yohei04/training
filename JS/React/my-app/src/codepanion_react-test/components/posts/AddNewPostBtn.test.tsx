import { fireEvent, render, screen } from '@testing-library/react';
import AddNewPostBtn from './AddNewPostBtn';

describe('AddNewPostBtn', () => {
  test('click button should trigger onClick callback', () => {
    const mockOnClick = jest.fn();
    render(<AddNewPostBtn onClick={mockOnClick} />);

    fireEvent.click(screen.getByText('Add New Post'));

    expect(mockOnClick).toHaveBeenCalled();
  });
});
