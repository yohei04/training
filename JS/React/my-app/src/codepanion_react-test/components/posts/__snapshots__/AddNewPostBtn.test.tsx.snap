// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewPostBtn <AddNewPostBtn /> matches snapshot 1`] = `
<DocumentFragment>
  <button>
    Add New Post
  </button>
</DocumentFragment>
`;
